neo/io/nwbio.py

```diff
@@ -0,0 +1,823 @@
+
+            if epoch_names is not None:
+                unique_epoch_names = np.unique(epoch_names)
+                for epoch_name in unique_epoch_names:
+                    index, = np.where((epoch_names == epoch_name))
+                    epoch = EpochProxy(self._file.epochs, epoch_name, index)
+                    if not lazy:
+                        epoch = epoch.load()
+                    segment_name = np.unique(segment_names[index])
+                    block_name = np.unique(block_names[index])
+                    assert segment_name.size == block_name.size == 1
+                    segment = self._get_segment(block_name[0], segment_name[0])
+                    segment.epochs.append(epoch)
+                    epoch.segment = segment
+            else:
+                epoch = EpochProxy(self._file.epochs)
+                if not lazy:
+                    epoch = epoch.load()
+                segment = self._get_segment("default", "default")
+                segment.epochs.append(epoch)
+                epoch.segment = segment
+
+    def _read_timeseries_group(self, group_name, lazy):
+        group = getattr(self._file, group_name)
+        for timeseries in group.values():
+            try:
+                # NWB files created by Neo store the segment and block names in the comments field
+                hierarchy = json.loads(timeseries.comments)
+            except JSONDecodeError:
+                # For NWB files created with other applications, we put everything in a single
+                # segment in a single block
+                # todo: investigate whether there is a reliable way to create multiple segments,
+                #       e.g. using Trial information
+                block_name = "default"
+                segment_name = "default"
+            else:
+                block_name = hierarchy["block"]
+                segment_name = hierarchy["segment"]
+            segment = self._get_segment(block_name, segment_name)
+            if isinstance(timeseries, AnnotationSeries):
+                event = EventProxy(timeseries, group_name)
+                if not lazy:
+                    event = event.load()
+                segment.events.append(event)
+                event.segment = segment
+            elif timeseries.rate:  # AnalogSignal
+                signal = AnalogSignalProxy(timeseries, group_name)
+                if not lazy:
+                    signal = signal.load()
+                segment.analogsignals.append(signal)
+                signal.segment = segment
+            else:  # IrregularlySampledSignal
+                signal = AnalogSignalProxy(timeseries, group_name)
+                if not lazy:
+                    signal = signal.load()
+                segment.irregularlysampledsignals.append(signal)
+                signal.segment = segment
+
+    def _read_units(self, lazy):
+        if self._file.units:
+            for id in self._file.units.id[:]:
```

Comment:
```suggestion
            for id in range(len(self._file.units)):
```

`id` is by default 0, 1, 2, 3, ... but that can be changed to any unique set of integers. I think this would be safer if you are going to use it to index a list